crates/api_common/src/site.rs

```diff
@@ -341,14 +346,27 @@ pub struct LeaveAdmin {
   pub auth: Sensitive<String>,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+pub struct BlockedInstance {
+  pub id: InstanceId,
+  pub domain: String,
+  pub published: chrono::NaiveDateTime,
+  pub updated: Option<chrono::NaiveDateTime>,
+  pub software: Option<String>,
+  pub reason: Option<String>,
+  pub version: Option<String>,
+}
+
 #[derive(Debug, Serialize, Deserialize, Clone)]
 #[cfg_attr(feature = "full", derive(TS))]
 #[cfg_attr(feature = "full", ts(export))]
 /// A list of federated instances.
 pub struct FederatedInstances {
   pub linked: Vec<Instance>,
   pub allowed: Vec<Instance>,
-  pub blocked: Vec<Instance>,
+  pub blocked: Vec<BlockedInstance>,
```

Comment:
No just return the Instance. The block and reasons should be included in the modlog only.

Reply:
this is to support the UI being able to show reasons for each blocked domain